### ISSUE TITLE
Incorrect warning about an unused import inside a macro syntax-quote

### ISSUE THREAD

**Issue by danielcompton** [borkdude/clj-kondo#706]
**version**: clj-kondo v2020.01.13
**platform**: macOS 10.14, native image via Homebrew
**problem**

clj-kondo warns about an unused import which is used as a type hint inside a syntax-quote in a macro.

**repro**

```clojure
(ns app.repro
  (:import java.lang.String))

(defmacro test-macro
  []
  ;^String (str "a" "b")
  `(let [test# ^String (str "a" "b")]
     test#))

#_(defn test-fn []
  ^String (str "a" "b"))
```

```console
 $ clj-kondo --lint src/clj/app/repro.clj
src/clj/app/repro.clj:2:12: warning: Unused import String
linting took 10ms, errors: 0, warnings: 1
```

**expected behavior**

No lint warning. If you uncomment `test-fn` or the line outside of the syntax-quote, you can see that clj-kondo correctly detects the import usage.

**Comment by borkdude** [comment]
@danielcompton Thanks. A workaround for now would be to fully qualify the type hint in the macro.

**Comment by borkdude** [comment]
@danielcompton Can you test with this new version?

https://8045-176829714-gh.circle-artifacts.com/0/release/clj-kondo-2020.01.14-SNAPSHOT-macos-amd64.zip

Preferably your entire codebase and not just this example to make sure no regressions were introduced.

**Comment by danielcompton** [comment]
I've just tested this, and can confirm that the fix worked and there are no regressions on our codebase. Thanks!
